apps/admin/src/pages/Users/Users.tsx

```diff
@@ -0,0 +1,304 @@
+import React, { useEffect } from 'react';
+import Select, { type GroupBase, type StylesConfig } from 'react-select';
+import { toast } from 'react-hot-toast';
+import { NextSeo } from 'next-seo';
+import Image from 'next/future/image';
+import classNames from 'classnames';
+import {
+  createColumnHelper,
+  getCoreRowModel,
+  SortingState,
+  useReactTable,
+} from '@tanstack/react-table';
+
+import {
+  useUpdateUserAsAdminMutation,
+  useGetAllUsersAdminLazyQuery,
+  SearchOrder,
+  Role,
+  type GetAllUsersAdminQuery,
+} from 'apollo-hooks';
+
+import Table from 'src/components/Table';
+
+import GithubIcon from '@/public/assets/github.svg';
+import useDebounce from 'src/components/Table/DebouncedInput';
+
+type Value = { value: string | number; label?: string };
+
+const options = [
+  {
+    value: Role.User,
+    label: Role.User,
+  },
+  {
+    value: Role.Admin,
+    label: Role.Admin,
+  },
+];
+
+const notifySuccess = (string: string) => toast.success(string);
+const notifyError = () => toast.error('Something went wrong');
+
+const styles: StylesConfig<Value, boolean, GroupBase<Value>> = {
+  indicatorSeparator: () => null,
+  control: (provided) => {
+    return {
+      ...provided,
+      width: '100%',
+      minWidth: '50px',
+      maxWidth: '120px',
+    };
+  },
+  menu: (base, { isDisabled }: any) => ({
+    ...base,
+    width: '100%',
+    minWidth: '50px',
+    maxWidth: '120px',
+    cursor: isDisabled ? 'not-allowed' : 'default',
+    background: isDisabled ? '#fff' : 'pointer',
+  }),
+  option: (provided, { isDisabled }) => {
+    return {
+      ...provided,
+      width: '100%',
+      minWidth: '50px',
+      maxWidth: '120px',
+      cursor: isDisabled ? 'not-allowed' : 'pointer',
+      background: isDisabled ? '#fff' : 'pointer',
+    };
+  },
+};
+
+const Users = () => {
+  const [updateUserAsAdmin] = useUpdateUserAsAdminMutation();
+  const [search, setSearch] = React.useState('');
+  const debouncedSearchTerm = useDebounce(search, 1000);
+  const [sorting, setSorting] = React.useState<SortingState>([
+    {
+      desc: false,
+      id: 'name',
+    },
+  ]);
+
+  const [getUsers, { data, loading, fetchMore }] = useGetAllUsersAdminLazyQuery(
+    {
+      notifyOnNetworkStatusChange: true,
+    }
+  );
+
+  useEffect(() => {
+    getUsers({
+      variables: {
+        input: {
+          search,
+          order: sorting?.[0]?.desc ? SearchOrder.Desc : SearchOrder.Asc,
+          orderBy: sorting?.[0]?.id,
+          cursor: data?.getAllUsersAdmin?.nextCursor,
+        },
+      },
+    });
+  }, [debouncedSearchTerm, sorting]);
+
+  const updateUser = async (user, role, action) => {
```

Comment:
make two separate functions, to update role and ban

Reply:
done